src/main/java/org/testng/TestRunner.java

```diff
@@ -544,9 +531,22 @@ private void initMethods() {
             true /* unique */,
             m_excludedMethods,
             comparator);
-    // shared group methods
-    m_groupMethods =
-        new ConfigurationGroupMethods(m_allTestMethods, beforeGroupMethods, afterGroupMethods);
+  }
+
+  private ITestNGMethod[] computeAndGetAllTestMethods() {
+    List<ITestNGMethod> testMethods = Lists.newArrayList();
+    for (ITestClass tc : m_classMap.values()) {
+      fixMethodsWithClass(tc.getTestMethods(), tc, testMethods);
+    }
+
+    return MethodHelper.collectAndOrderMethods(
```

Comment:
Proposal:

```java
ITestNGMethod[] result = MethodHelper.collectAndOrderMethods(testMethods,
        true /* forTest? */,
        m_runInfo,
        m_annotationFinder,
        false /* unique */,
        m_excludedMethods,
        comparator);
if (!RuntimeBehavior.isMemoryFriendlyMode()) {
   m_allTestMethods = result;
}
return result;
```

Reply:
This is not needed. Please have a look at the new changes that I pushed in and we can further discuss based on that.